sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/AggregatePushDownUtils.scala

```diff
@@ -52,8 +52,10 @@ object AggregatePushDownUtils {
 
     def processMinOrMax(agg: AggregateFunc): Boolean = {
       val (column, aggType) = agg match {
-        case max: Max => (max.column, "max")
-        case min: Min => (min.column, "min")
+        case max: Max if max.column().isInstanceOf[NamedReference] =>
+          (max.column.asInstanceOf[NamedReference], "max")
+        case min: Min if min.column().isInstanceOf[NamedReference] =>
+          (min.column.asInstanceOf[NamedReference], "min")
         case _ =>
           throw new IllegalArgumentException(s"Unexpected type of AggregateFunc ${agg.describe}")
```

Comment:
does the caller side guarantee to pass an `AggregateFunc` with only `Min`/`Max` and `NamedReference`?